test/suggestion.cpp

```diff
@@ -0,0 +1,101 @@
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#include <zim/zim.h>
+#include <zim/archive.h>
+#include <zim/search.h>
+#include <zim/writer/creator.h>
+#include <zim/writer/item.h>
+#include <zim/writer/contentProvider.h>
+
+#include "tools.h"
+
+#include "gtest/gtest.h"
+
+namespace {
+
+  class TestItem : public zim::writer::BasicItem {
+      public:
+      TestItem(const std::string& path, const std::string& mimetype, const std::string& title)
+        : BasicItem(path, mimetype, title) {}
+      
+      virtual std::unique_ptr<zim::writer::ContentProvider> getContentProvider() const {
+        return std::unique_ptr<zim::writer::ContentProvider>(new zim::writer::StringProvider(""));
+      }
+  };
+
+  std::string createTestZim(std::vector<std::string> titles) {
+    zim::unittests::TempFile temp("testZim");
+    auto tempPath = temp.path() + ".zim";
+    
+    zim::writer::Creator creator;
+    creator.configIndexing(true, "eng");
+    creator.startZimCreation(tempPath);
+
+    // add dummy items with given titles
+    for (auto title : titles) {
+      std::string path = "dummyPath" + title;
+      auto item = std::make_shared<TestItem>(path, "plain/text", title);
+      creator.addItem(item);
+    }
+
+    creator.addMetadata("Title", "This is a title");
+    creator.setMainPath("foo");
+    creator.finishZimCreation();
+
+    return tempPath;
+  }
+
+  std::vector<std::string> getSuggestions(const zim::Archive archive, std::string query) {
+    zim::Search search(archive);
+    search.set_suggestion_mode(true);
+    search.set_query(query);
+    search.set_range(0, archive.getEntryCount());
+    search.set_verbose(true);
+    
+    std::vector<std::string> result;
+    for (auto entry = search.begin();entry!=search.end();entry++) {
+      result.push_back((*entry).getTitle());
+    }
+    
+    return result;
+  }
+
+  TEST(Suggestion, singleTermQuery) {
+    std::vector<std::string> titles = {
+                                        "berlin", 
+                                        "hotel berlin, berlin", 
+                                        "not berlin", 
+                                        "berlin wall", 
+                                        "again berlin", 
+                                        "fooland"
+                                       };
+    
+    std::string path = createTestZim(titles);
+    const zim::Archive archive(path);
+    std::vector<std::string> resultSet = getSuggestions(archive, "berlin");
+    
+    ASSERT_EQ(titles.size() - 1, resultSet.size());
+    
+    for(int i = 0;(unsigned int)i < resultSet.size();i++){
+      std::cout<<titles[i]<<", "<<resultSet[i]<<std::endl;
+      ASSERT_EQ(titles[i], resultSet[i]);
+    }
```

Comment:
It would be better if the order in which the items are added is different from the expected order of the query results. Otherwise, the matching order may have a much simpler explanation.

Reply:
Right, Shuffling the order.